providers/nodesmith-provider.d.ts

```diff
@@ -0,0 +1,8 @@
+import { JsonRpcProvider, JsonRpcSigner } from './json-rpc-provider';
+import { Networkish } from '../utils/networks';
+export declare class NodesmithProvider extends JsonRpcProvider {
+    readonly apiKey: string;
+    constructor(apiKey: string, network?: Networkish);
```

Comment:
Yes, regardless hows it would be accepted. Network should be a first param, and apiKey shiuld be the second, as to have only one provider with reversed params is really adds misconception.